kernel/rule.ml

```diff
@@ -17,28 +17,30 @@ type wf_pattern =
 
 type rule_name = Delta of name | Gamma of bool * name
 
+type condition = {left:term; right:term}
+
 type 'a rule =
   {
     name: rule_name;
     ctx: 'a;
     pat: pattern;
+    cond: condition option;
```

Comment:
From what I understand from the rest of the code, we could have a list of conditions (interrpreted as their conjunction) here.

Reply:
This is done in my last commit.

Reply:
Nice!